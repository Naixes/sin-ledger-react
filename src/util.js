export const LIST_VIEW = 'list'
export const CHART_VIEW = 'chart'
export const TYPE_INCOME = 'income'
export const TYPE_OUTCOME = 'outcome'

export const range = (size, startAt) => {
  let arr = []
  for (let index = 0; index < size; index++) {
    arr[index] = startAt + index    
  }
  return arr
}

export const Colors = {
  blue: '#347eff',
  deepBlue: '#61dafb',
  green: '#28a745',
  red: '#dc3545',
  gray: '#555',
  lightGray: '#efefef',
  white: '#fff',
}