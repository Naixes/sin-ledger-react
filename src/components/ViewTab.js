import React from 'react';
import PropTypes from 'prop-types'
import Ionicon from 'react-ionicons'
import {Colors, LIST_VIEW, CHART_VIEW} from '../util'

const generateLinkClass = (activeTab, current) => {
    return (activeTab === current) ? 'active nav-link' : 'nav-link'
}

const ViewTab = ({activeTab, onChangeTab}) => {
    return (
        <ul className="nav nav-tabs nav-fill my-4">
            <li className="nav-item">
                {/* eslint-disable-next-line */}
                <a 
                    className={generateLinkClass(activeTab, LIST_VIEW)} 
                    href='#'
                    onClick={(e) => {e.preventDefault(); onChangeTab(LIST_VIEW)}}
                >
                    <Ionicon
                        // mr：margin-right
                        className="rounded-circle mr-2"
                        fontSize="25px"
                        color={Colors.blue}
                        icon='ios-paper'
                    ></Ionicon>
                    列表模式
                </a>
            </li>
            <li className="nav-item">
                {/* eslint-disable-next-line */}
                <a 
                    className={generateLinkClass(activeTab, CHART_VIEW)}
                    href='#'
                    onClick={(e) => {e.preventDefault(); onChangeTab(CHART_VIEW)}}
                >
                    <Ionicon
                        className="rounded-circle mr-2"
                        fontSize="25px"
                        color={Colors.blue}
                        icon='ios-pie'
                    ></Ionicon>
                    图表模式
                </a>
            </li>
        </ul>
    )
}

ViewTab.propTypes = {
    activeTab: PropTypes.string.isRequired,
    onChangeTab: PropTypes.func.isRequired
}

export default ViewTab