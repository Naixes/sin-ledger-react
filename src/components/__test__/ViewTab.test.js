import React from 'react'
import Ionicon from 'react-ionicons'

import { shallow } from 'enzyme'
import ViewTab from '../ViewTab'
import {LIST_VIEW} from '../../util'

const props = {
    activeTab: LIST_VIEW,
    onChangeTab: jest.fn()
}

describe('test ViewTab component', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<ViewTab {...props}/>)
    })
    it('render the component to match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
});
