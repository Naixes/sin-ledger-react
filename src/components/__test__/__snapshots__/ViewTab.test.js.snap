// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ViewTab component render the component to match snapshot 1`] = `ShallowWrapper {}`;
